app/services/fetch_csv_service.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require 'open-uri'
+require 'net/http'
+require 'byebug'
+
+module Services
+  class FetchCsvService
+    def initialize(url)
+      @url = url
+    end
+
+    def call
+      uri = URI.parse(@url)
+      response = Net::HTTP.get_response(uri)
+      raise "#{response.code} #{response.message}" unless response.is_a?(Net::HTTPOK)
+
+      response.body
+    rescue StandardError => e
+      raise "Error fetching CSV: #{e.message}"
```

Comment:
Your Roda setup will catch this and give it right to the end user. Should the end user be aware/responsible on handling all server errors?

Reply:
End users should never see raw server errors.
It would be better to log all standard errors and throw just some specific ones with a generic message.
For exampel:

```
rescue StandardError => e
  log_error("Error fetching CSV: #{e.message}")
rescue URI::InvalidURIError => e
  log_error("Invalid URL: #{@url}")
  raise "Invalid URL format"
```